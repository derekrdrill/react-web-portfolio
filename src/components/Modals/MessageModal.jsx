import React, { useContext } from 'react';
import styled from 'styled-components';
import { Button, Typography, Modal } from '@mui/material';
import { Container, Row, Col } from 'react-bootstrap';

import { DarkLightModeContext } from '../DarkLightMode/context/DarkLightModeContext';

export const MessageModal = ({
  open,
  onClose,
  onSubmit,
  title,
  body,
  cancelButtonText,
  confirmButtonText,
  cancelButtonColor,
  confirmButtonColor,
}) => {
  const { darkMode } = useContext(DarkLightModeContext);

  return (
    <Modal open={open} onClose={onClose}>
      <ModalContainer darkMode={darkMode}>
        <ModalRow>
          <Typography variant='h6' component='h1' className='text-center'>
            {title}
          </Typography>
        </ModalRow>
        <ModalRow>
          <Typography variant='body1' component='p' className='text-center'>
            {body}
          </Typography>
        </ModalRow>
        <ModalRow>
          <ModalButtonCol xs={6}>
            <Button
              color={darkMode ? 'secondary' : cancelButtonColor || 'error'}
              variant='contained'
              onClick={onClose}
              fullWidth
            >
              {cancelButtonText}
            </Button>
          </ModalButtonCol>
          <ModalButtonCol xs={6}>
            <Button
              color={darkMode ? 'primary' : confirmButtonColor || 'success'}
              variant='contained'
              onClick={onSubmit}
              fullWidth
            >
              {confirmButtonText}
            </Button>
          </ModalButtonCol>
        </ModalRow>
      </ModalContainer>
    </Modal>
  );
};

const ModalContainer = styled(Container)(({ darkMode }) => ({
  'h1, h2, h3, h4, h5, h6, p': {
    color: darkMode && 'black',
  },
  backgroundColor: darkMode ? 'grey' : 'white',
  marginLeft: '32%',
  marginTop: '18%',
  maxWidth: '35%',
  padding: 30,
  borderRadius: 5,
}));

const ModalRow = styled(Row)({
  padding: 10,
});

const ModalButtonCol = styled(Col)({
  textAlign: 'center',
});
