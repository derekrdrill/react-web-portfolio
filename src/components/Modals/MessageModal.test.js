import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { MessageModal, ModalContainer } from './MessageModal';
import { getCancelButtonColor, getConfirmButtonColor } from './MessageModal';

configure({ adapter: new Adapter() });

const renderer = new ShallowRenderer();

describe('Message Modal tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({ darkMode: false });
    renderer.render(<MessageModal open />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders ModalContainer styled component correctly', () => {
    const modalContainer = shallow(<ModalContainer />);
    const modalContainerDark = shallow(<ModalContainer darkMode />);

    expect(modalContainer.props().className).toEqual('sc-beySbM gtkpVX');
    expect(modalContainerDark.props().className).toEqual('sc-beySbM fnAFBm');
  });

  it('runs getCancelButtonColor', () => {
    expect(getCancelButtonColor('#FFF', true)).toEqual('secondary');
    expect(getCancelButtonColor('#FFF', false)).toEqual('#FFF');
    expect(getCancelButtonColor(null, false)).toEqual('error');
  });

  it('runs getConfirmButtonColor', () => {
    expect(getConfirmButtonColor('#FFF', true)).toEqual('primary');
    expect(getConfirmButtonColor('#FFF', false)).toEqual('#FFF');
    expect(getConfirmButtonColor(null, false)).toEqual('success');
  });
});
