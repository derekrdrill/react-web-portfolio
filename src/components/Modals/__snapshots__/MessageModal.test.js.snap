// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Modal tests renders correctly 1`] = `
<ForwardRef(Modal)
  open={true}
>
  <Styled(Container)
    darkMode={false}
  >
    <Styled(Row)>
      <ForwardRef(Typography)
        className="text-center"
        component="h1"
        variant="h6"
      />
    </Styled(Row)>
    <Styled(Row)>
      <ForwardRef(Typography)
        className="text-center"
        component="p"
        variant="body1"
      />
    </Styled(Row)>
    <Styled(Row)>
      <Styled(Col)
        sm={
          Object {
            "offset": 3,
            "span": 3,
          }
        }
        xs={6}
      >
        <ForwardRef(Button)
          color="error"
          fullWidth={true}
          variant="contained"
        />
      </Styled(Col)>
      <Styled(Col)
        sm={
          Object {
            "span": 3,
          }
        }
        xs={6}
      >
        <ForwardRef(Button)
          color="success"
          fullWidth={true}
          variant="contained"
        />
      </Styled(Col)>
    </Styled(Row)>
  </Styled(Container)>
</ForwardRef(Modal)>
`;
